kds/kds.go

```diff
@@ -692,22 +692,51 @@ func ProductName(product *pb.SevProduct) string {
 	if product == nil {
 		product = abi.DefaultSevProduct()
 	}
-	return fmt.Sprintf("%s-%02X", ProductString(product), product.ModelStepping)
+	model := (product.ModelStepping >> 4) & 0xf
+	if model != 1 {
+		return "Unknown"
+	}
+	stepping := product.ModelStepping & 0xf
+	// The string is not fully programmable, given the processor programming reference revisions.
+	switch product.Name {
+	case pb.SevProduct_SEV_PRODUCT_MILAN:
+		return fmt.Sprintf("Milan-B%X", stepping)
```

Comment:
Just out of curiosity: I saw Milan-B0/Genoa-A0/etc. Is it documented in the spec that Milan revision will always start with B and Genoa revision will always start with A?

Reply:
I specifically don't know for sure. I'm using the A%X naming provisionally given the Zen4 silicon revision was A0 last year, but I see B0 could be coming or is already here. I'm unsure how to differentiate between the two and I've asked AMD what the deal is. It might be model == 0 => "A", and model == 1 => "B", but I don't know.

Reply:
I checked in with one of our KVM engineers and learned the stepping <-> stepping version translation is not algorithmic, but just a matter of manufacturer decision whenever they change the CPUID stepping value. Updated the logic accordingly with lookup tables given information known about taped out stepping versions.